lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -190,6 +199,26 @@ export default class BitcoinProcessor {
     void this.periodicPoll();
   }
 
+  private async upgradeDatabaseIfNeeded () {
+    const currentServiceVersion = await this.getServiceVersion();
+    const savedServiceState = await this.serviceStateStore.get();
+    const savedServiceVersion = savedServiceState ? savedServiceState.serviceVersion : 'unknown';
+
+    // Upgrade the DB if the saved service version is different to current running service version.
+    if (savedServiceVersion !== currentServiceVersion.version) {
```

Comment:
So every time we release a new npm package it will trigger this upgrade right? If yes, then I'm not sure if that is what we want.

Reply:
Happy to discuss further. I am not suggesting that this is how we DB upgrade in the long run, but my stance is that this is fine for now given how little data we have right now. While wiping away DB is the dumbest way to DB upgrade, it is "a way that works". Keep in mind that we have the opportunity to, and we should revisit this DB upgrade code every time we release a new version.

Reply:
Discussed offline, will target upgrade only on DB with unknown service version.